goaway.go

```diff
@@ -143,62 +143,63 @@ func (g *ProfanityDetector) indexToRune(s string, index int) int {
 	count := 0
 	for i := range s {
 		if i == index {
-			return count
+			break
+		}
+		if i < index {
+			count++
 		}
-		count++
 	}
-	return -1
+	return count
 }
 
 // Censor takes in a string (word or sentence) and tries to censor all profanities found.
 func (g *ProfanityDetector) Censor(s string) string {
 	censored := []rune(s)
 	var originalIndexes []int
 	s, originalIndexes = g.sanitize(s, true)
-	// Check for false negatives
-	for _, word := range g.falseNegatives {
-		currentIndex := 0
-		for currentIndex != -1 {
-			if foundIndex := strings.Index(s[currentIndex:], word); foundIndex != -1 {
-				for i := 0; i < len([]rune(word)); i++ {
-					runeIndex := g.indexToRune(string(censored), currentIndex+foundIndex+i)
-					censored[originalIndexes[runeIndex]] = '*'
+	runeWordLength := 0
+
+	checkProfanity := func(wordList []string) {
+		for _, word := range wordList {
+			currentIndex := 0
+			runeWordLength = len([]rune(word))
+			for currentIndex != -1 {
+				if foundIndex := strings.Index(s[currentIndex:], word); foundIndex != -1 {
+					for i := 0; i < runeWordLength; i++ {
+						runeIndex := g.indexToRune(s, currentIndex+foundIndex) + i
+						if runeIndex < len(originalIndexes) {
+							censored[originalIndexes[runeIndex]] = '*'
+						}
+					}
+					currentIndex += foundIndex + len([]byte(word))
+				} else {
+					break
```

Comment:
Can you make a separate method for `checkProfanity` and `removeFalsePositives` specifically for `Censor` instead of creating a function on-the-fly?